esignet-core/src/main/java/io/mosip/esignet/core/dto/ApiRateLimit.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package io.mosip.esignet.core.dto;
+
+import lombok.Data;
+
+import java.io.Serializable;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Data
+public class ApiRateLimit implements Serializable {
+
+    ConcurrentHashMap<Integer, Integer> count = new ConcurrentHashMap<>();
+    ConcurrentHashMap<Integer, Long> lastInvocation = new ConcurrentHashMap<>();
+
+    public void increment(int apiCode) {
+        count.compute(apiCode, (k, v) -> (v == null) ? 1 : v + 1);
+        lastInvocation.compute(apiCode, (k, v) -> System.currentTimeMillis());
```

Comment:
I was not able to find, where this timing part is used.

Reply:
Also, time gap related requirements from the ticket is missing.

"The number of times a request is allowed to be invoked per individualID/transaction and the time gap between each request(human or bot distinction) should be considered.

Each auth factor should have different considerations. For example, OTP is just to type the value as received on email/phone so the number of times it can be mistaken is less compared to biometric auth where many external factors need to be right to get a good capture.

For requests that come from eSignet where a captcha token is used to differentiate between human and robot actions already, the allowed number of attempts and the time gap between the attempts need to be different. It means that IDA auth endpoint and delegated kyc-auth endpoint should be configurable with different attempt counts and time gap b/w attempts."

Reply:
For now i have added time gap check between requests.